src/safe_cli/tx_builder/tx_builder_file_decoder.py

```diff
@@ -0,0 +1,240 @@
+
+        raise SoliditySyntaxError("Invalid Boolean value")
+
+    return bool(value)
+
+
+def parse_int_value(value: str) -> int:
+    trimmed_value = value.replace('"', "").replace("'", "").strip()
+
+    if trimmed_value == "":
+        raise SoliditySyntaxError("Invalid empty strings for integers")
+    try:
+        if not trimmed_value.isdigit() and bool(
+            re.fullmatch(r"0[xX][0-9a-fA-F]+|[0-9a-fA-F]+$", trimmed_value)
+        ):
+            return int(trimmed_value, 16)
+
+        return int(trimmed_value)
+    except ValueError:
+        raise SoliditySyntaxError("Invalid integer value")
+
+
+def parse_string_to_array(value: str) -> List[Any]:
+    number_of_items = 0
+    number_of_other_arrays = 0
+    result = []
+    value = value.strip()[1:-1]  # remove the first "[" and the last "]"
+
+    for char in value:
+        if char == "," and number_of_other_arrays == 0:
+            number_of_items += 1
+            continue
+
+        if char == "[":
+            number_of_other_arrays += 1
+        elif char == "]":
+            number_of_other_arrays -= 1
+
+        if len(result) <= number_of_items:
+            result.append("")
+
+        result[number_of_items] += char.strip()
+
+    return result
+
+
+def _get_base_field_type(field_type: str) -> str:
+    trimmed_value = field_type.strip()
+    if not trimmed_value:
+        raise SoliditySyntaxError("Empty base field type for")
+
+    base_field_type_regex = re.compile(
+        r"^([a-zA-Z0-9]*)(((\[\])|(\[[1-9]+[0-9]*\]))*)?$"
+    )
+    match = base_field_type_regex.match(trimmed_value)
+    if not match:
+        raise SoliditySyntaxError(f"Unknown base field type from {trimmed_value}")
+    return match.group(1)
+
+
+def _is_array(values: str) -> bool:
+    trimmed_value = values.strip()
+    return trimmed_value.startswith("[") and trimmed_value.endswith("]")
+
+
+def parse_array_of_values(values: str, field_type: str) -> List[Any]:
+    if not _is_array(values):
+        raise SoliditySyntaxError("Invalid Array value")
+
+    parsed_values = parse_string_to_array(values)
+    return [
+        (
+            parse_array_of_values(item_value, field_type)
+            if _is_array(item_value)
+            else parse_input_value(_get_base_field_type(field_type), item_value)
+        )
+        for item_value in parsed_values
+    ]
+
+
+def is_boolean_field_type(field_type: str) -> bool:
+    return field_type == "bool"
+
+
+def is_int_field_type(field_type: str) -> bool:
+    return field_type.startswith("uint") or field_type.startswith("int")
+
+
+def is_tuple_field_type(field_type: str) -> bool:
+    return field_type.startswith("tuple")
+
+
+def is_bytes_field_type(field_type: str) -> bool:
+    return field_type.startswith("bytes")
+
+
+def is_array_of_strings_field_type(field_type: str) -> bool:
+    return field_type.startswith("string[")
+
+
+def is_array_field_type(field_type: str) -> bool:
+    pattern = re.compile(r"\[\d*\]$")
```

Comment:
There's no need to escape the `]` here.

```suggestion
    pattern = re.compile(r"\[\d*]$")
```